src/python/pants/backend/jvm/tasks/jvm_compile/jvm_compile.py

```diff
@@ -626,6 +651,31 @@ def joined_dep_msg(deps):
         self.context.log.warn('Target {} had {}\n'.format(
           compile_context.target.address.spec, unused_msg))
 
+  def _find_missing_deps(self, compile_failure_log, target):
+    with self.context.new_workunit('missing-deps-suggest', labels=[WorkUnitLabel.COMPILER]):
+      missing_dep_suggestions, no_suggestions = self._missing_deps_finder.find(
+        compile_failure_log, target)
+
+      if missing_dep_suggestions:
+        self.context.log.info('Found the following deps from target\'s transitive '
+                              'dependencies that contain the not found classes:')
+        suggested_deps = set()
+        for classname, candidates in missing_dep_suggestions.items():
+          suggested_deps.add(list(candidates)[0])
+          self.context.log.info('  {}: {}'.format(classname, ', '.join(candidates)))
+        suggestion_msg = (
+          '\nIf the above information is correct, '
+          'please add the following to the dependencies of ({}):\n  {}\n'
+            .format(target.address.spec, '\n    '.join(sorted(list(suggested_deps))))
+        )
+        self.context.log.info(suggestion_msg)
+
+      if no_suggestions:
```

Comment:
IMO, I don't think it makes sense to log the case where you don't find anything (that seems obvious enough). Maybe you could do it at debug level though?